zebrad/tests/acceptance.rs

```diff
@@ -1706,6 +1706,57 @@ async fn delete_old_databases() -> Result<()> {
     Ok(())
 }
 
+#[tokio::test]
+async fn stored_config_works() -> Result<()> {
+    zebra_test::init();
+
+    let run_dir = testdir()?;
+    let stored_config_path = stored_config_path();
+
+    // run zebra with stored config
+    let mut child =
+        run_dir.spawn_child(args!["-c", stored_config_path.to_str().unwrap(), "start"])?;
+
+    //zebra was able to start with the stored config
+    child.expect_stdout_line_matches("Starting zebrad".to_string())?;
+
+    // finish
+    child.kill()?;
+
+    let output = child.wait_with_output()?;
+    let output = output.assert_failure()?;
+
+    // [Note on port conflict](#Note on port conflict)
+    output
+        .assert_was_killed()
+        .wrap_err("Possible port conflict. Are there other acceptance tests running?")?;
+
+    Ok(())
+}
+
+#[tokio::test]
+async fn stored_config_is_newest() -> Result<()> {
+    zebra_test::init();
+
+    let run_dir = testdir()?;
+    let stored_config_path = stored_config_path();
+    let generated_config_path = run_dir.path().join("newest_config.toml");
+
+    // generate an up to date config
+    let child =
+        run_dir.spawn_child(args!["generate", "-o": generated_config_path.to_str().unwrap()])?;
+
+    let output = child.wait_with_output()?;
+    let _output = output.assert_success()?;
+
+    let contents_generated = std::fs::read_to_string(generated_config_path).unwrap();
+    let contents_stored = std::fs::read_to_string(stored_config_path).unwrap();
+
+    assert_eq!(contents_generated, contents_stored);
```

Comment:
We might want to tell developers why the test failed, and what command they need to run to update the stored config file.

Reply:
This is a good one, thanks. Added in https://github.com/ZcashFoundation/zebra/pull/4676/commits/f423afbcf3be61989e29bcaccc833042c1ba967a